src/bitcoin-cli.cpp

```diff
@@ -191,7 +192,95 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
 }
 #endif
 
-static UniValue CallRPC(const std::string& strMethod, const UniValue& params)
+/** Class that handles the conversion from a command-line to a JSON-RPC request,
+ * as well as converting back to a JSON object that can be shown as result.
+ */
+class BaseRequestHandler
+{
+public:
+    virtual UniValue PrepareRequest(const std::string& strMethod, const std::vector<std::string>& args) = 0;
+    virtual UniValue ProcessReply(const UniValue &batch_in) = 0;
+};
+
+/** Process getinfo requests */
+class GetinfoRequestHandler: public BaseRequestHandler
+{
+public:
+    const int ID_NETWORKINFO = 0;
+    const int ID_BLOCKCHAININFO = 1;
+    const int ID_WALLETINFO = 2;
+
+    /** Create a simulated `getinfo` request. */
+    UniValue PrepareRequest(const std::string& strMethod, const std::vector<std::string>& args) override
```

Comment:
nit: function arguments should not be hungarian/camel case `strMethod` -> `method`

Reply:
> nit: function arguments should not be hungarian/camel case strMethod -> method

not currently addressed

Reply:
Done